tools/webidl_binder.py

```diff
@@ -344,20 +333,25 @@ def type_to_cdec(raw):
     return ret
   return ret + '*'
 
-def render_function(class_name, func_name, sigs, return_type, non_pointer, copy, operator, constructor, func_scope, call_content=None, const=False, array_attribute=False):
+def render_function(class_name, func_name, sigs, return_type, non_pointer, copy, operator, constructor, func_scope, call_content=None, const=False, sequence_attribute=False):
   global mid_c, mid_js, js_impl_methods
 
+  if return_type is None:
+    return_type = WebIDL.BuiltinTypes[WebIDL.IDLBuiltinType.Types.void]
+
   legacy_mode = CHECKS not in ['ALL', 'FAST']
   all_checks = CHECKS  == 'ALL'
 
   bindings_name = class_name + '_' + func_name
   min_args = min(sigs.keys())
   max_args = max(sigs.keys())
+  if return_type.isSequence():
+    return_type = return_type.inner
```

Comment:
these lines look surprising - is it saying that a function returning `sequence<long>` returns `long`? Or does `inner` mean something else?

Reply:
Ah, very good point!  I didn't think that webidl bindings supported returning array values, and that attribute getters for arrays were the only case.  I was wrong.  I was confused by the fact that the getter for `render_function` was called with the array type's `.name` property; but of course this is without the `[]`.  Instead for array (or rather, sequence) attribute getters I should call `render_function` with the sequence's `.inner.type` and remove this case.